pkg/clusterrouter/proxy/proxy_server.go

```diff
@@ -227,6 +231,13 @@ func (r *RemoteService) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		er.Error(rw, req, err)
 		return
 	}
+
+	if httpstream.IsUpgradeRequest(req) {
```

Comment:
@kinarashah  What's the purpose of this code?

Reply:
It's to check if it's a request that has `Connection`: `Upgrade` in its headers, so this applies only to websocket requests